assets/src/components/form/inputs/date-time-input.js

```diff
@@ -84,162 +46,112 @@ class DateTimeDropdown extends Component {
 				// Reverse the string and test for "a" not followed by a slash
 				.split( '' ).reverse().join( '' )
 		);
-		this.setState( {
-			is12HourTime,
-			editorOpen: false,
-			inputValue: initialValue,
-			newValue: initialValue,
-			prevValue: initialValue,
-			onChange: this.props.input && this.props.input.onChange ?
-				this.props.input.onChange :
-				null,
-		} );
-	};
+		setIs12HourTime( _is12HourTime );
+	}, [] );
 
-	/**
-	 * opens and closes DateTimePicker Modal
-	 *
-	 * @function
-	 */
-	toggleEditor = () => {
-		this.setState( ( prevState ) => (
-			{ editorOpen: ! prevState.editorOpen }
-		) );
-	};
+	const {
+		input,
+		htmlId,
+		htmlClass,
+		helpTextID,
+		dataSet,
+		inputWidth = '',
+		...attributes
+	} = props;
+	delete attributes.initialValue;
```

Comment:
can we plz move the destructuring to the function declaration?

ie:

```jsx
const DateTimeDropdown = ( {
	input,
	htmlId,
	htmlClass,
	helpTextID,
	initialValue,
	dataSet,
	inputWidth = '',
	...attributes
} ) => {
```

plz note that I also included `initialValue` as one of the props so that we don't have to delete it from the attributes afterwards (no idea why I didn't do that initially?!?!)

I think this will work better as it allows IDEs to detect the props for the component

this of course will also require you to change any place where you access props via: `props.{prop name}`

Reply:
Moved to function declaration. Now it looks good :)